### PR TITLE
Change Mocha to not need a babel build to run

### PR DESCRIPTION
**Pull request by darrenhaken**
Changed the Gulp test run so that it no longer requires Babel to transpile the ES6 code before running. Instead, Babel is fed into Mocha as a compiler and it happens on the fly in memory. 

I have also added a watch task which should allow you to run the unit tests and then monitor for changes (and run the diff) which helps with a TDD workflow.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jiahaog) [inline .babelrc]: Do we need this? I thought it's already in the `package.json`?

File: test/module/options/normalizeUrlSpec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import 'source-map-support/register';
```
Comment (jiahaog) [inline test/module/options/normalizeUrlSpec.js]: Do we have to import this file? I can't really remember if it's needed for all files

File: package.json [hunk 3]
```diff
@@ -13,8 +13,9 @@
   "scripts": {
     "dev-up": "npm install && (cd app && npm install) && npm run build",
     "test": "gulp test",
+    "test-watch": "mocha --watch --require babel-register --recursive ./test",
```
Comment (jiahaog) [inline package.json]: Can't really remember how gulp works >< Is this related to the `tdd` gulp task above?

File: package.json [hunk 4]
```diff
@@ -13,8 +13,9 @@
...
     "lint": "eslint .",
-    "ci": "gulp test && npm run lint",
+    "ci": "gulp build test && npm run lint",
```
Comment (jiahaog) [inline package.json]: Is this change needed?

File: .babelrc (hunk 1 above)
Comment (darrenhaken) [inline .babelrc]: I didn't realise we had it in the package.json, I'll remove the file. Good spot!

File: test/module/options/normalizeUrlSpec.js (hunk 2 above)
Comment (darrenhaken) [inline test/module/options/normalizeUrlSpec.js]: I wasn't sure myself but I left it in whilst I wasn't fully aware what was needed to run tests with the existing setup. I've removed it and all seems fine so I can take this out

File: package.json (hunk 4 above)
Comment (darrenhaken) [inline package.json]: This I missed and it should actually be now calling `gulp tdd`. I'll push a change now and you can see.

**Comment by darrenhaken** [comment]
@jiahaog please review new changes based on comments

**Comment by darrenhaken** [comment]
Awesome that's all merged in then
